File: 167/solution.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        j = len(numbers) - 1
        i = 0
        while i < j:
            tmp = numbers[i] + numbers[j]
            if tmp == target:
                return [i + 1, j + 1]
            elif tmp > target:
                j -= 1
            else:
                i += 1
        return []

File: 167/solution.go
package _167

func twoSum(numbers []int, target int) []int {
	i := 0
	j := len(numbers) - 1
	for i < j {
		tmp := numbers[i] + numbers[j]
		if tmp == target {
			return []int{i + 1, j + 1}
		} else if tmp > target {
			j--
		} else {
			i++
		}
	}
	return []int{}
}
